iterating-over-data/practice-iterating-over-data.md

```diff
@@ -0,0 +1,465 @@
+# Practice: Iterating Over Data
+
+## Directions
+
+Complete all questions below.
+
+## Practice
+
+<!-- Question 1 -->
+<!-- prettier-ignore-start -->
+### !challenge
+
+* type: multiple-choice
+* id: EJew-RW0t
+* title: Which Range is Correct?
+
+##### !question
+
+Given a function called `stopwatch` that takes in the argument `22`, which of the following loops would  correctly start at `1` and increment all the way up to *and include* `22`?
+
+##### !end-question
+
+##### !options
+
+* `for i in 22:`
+* `for i in range(22)`
+* `for i in range(1,22)`
+* `for i in range(1,23)`
+
+##### !end-options
+
+##### !answer
+
+`for i in range(1,23)`
+
+##### !end-answer
+
+
+##### !hint
+
+##### !end-hint
+
+
+##### !rubric
+
+##### !end-rubric
+
+
+##### !explanation
+
+Python's `range` has an *exclusive nature*, meaning that it does not include the last integer in its range, ie the `stop` parameter. That's why we must use `23` instead of just `22`.
+
+##### !end-explanation
+
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 2 -->
+<!-- prettier-ignore-start -->
+### !challenge
+
+* type: code-snippet
+* language: python3.6
+* id: EkdvU6Z0K
+* title: Iterate Over a List
+
+##### !question
+
+Create a function named `sum_list` that takes a parameter `list` (a list of integers). Iterate through the list and sum up each integer. Then return the total.
+
+##### !end-question
+
+##### !placeholder
+
+```python
+
+def sum_list(list):
+  pass
```

Comment:
Learn does syntax highlighting with the keyword `list`, which could be really jarring (in the past, Ada students have been really confused at what gets highlighted and why it does, taking up much conversation)

I suggest either changing the var name `list`, or adding a quick explanation.

In another lesson, I've added this note.

```suggestion
def sum_list(list):
    # Note: the word "id" may be highlighted because the Learn platform believes it's reserved
    # This should not affect our code
    pass
```

Also, we should conform to 4 space indents